docs/msrv_policy.md

```diff
@@ -0,0 +1,22 @@
+# Minimum Supported Rust Versions policy
+
+The diplomat libraries that end up as dependencies of diplomat-using crates (`diplomat`, `diplomat_core`, `diplomat-runtime`) attempt to maintain a reasonable Minimum Supported Rust Version policy for its users.
```

Comment:
observation: `diplomat-core` is not a direct dependency of downstream crates. It should suffice to put `diplomat` and `diplomat-runtime` under MSRV, which has the side effect of excluding HIR from MSRV, as `diplomat` doesn't use it.